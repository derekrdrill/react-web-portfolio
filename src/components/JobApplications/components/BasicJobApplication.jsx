import React, { useContext, useState } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { Grid, TextField, Typography, Button } from '@mui/material';
import { DynamicList } from '../../DynamicList/DynamicList';
import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { formInputsGenerator } from '../../../utils/formInputsGenerator';

import { BASIC_JOB_APP_INPUTS } from '../constants/BASIC_JOB_APP_INPUTS';
import { DynamicFormInputs } from '../../DynamicFormInputs/DynamicFormInputs';

export const BasicJobApplication = () => {
  const { darkMode } = useContext(DarkLightModeContext);
  const [form, setForm] = useState(formInputsGenerator(BASIC_JOB_APP_INPUTS));

  return (
    <JobAppContainer darkMode={darkMode} container spacing={2}>
      <JobAppTitleContainer container>
        <Grid item xs={12}>
          <JobAppTitleText darkMode={darkMode} variant='h4'>
            Application for Employment
          </JobAppTitleText>
        </Grid>
      </JobAppTitleContainer>
      <FileUploadContainer darkMode={darkMode} item xs={12}>
        <Typography component='h5'>Upload supporting documents</Typography>
        <FileUploadScrollContainer container darkMode={darkMode}>
          <DynamicList
            addColor='forestgreen'
            removeColor='maroon'
            children={<FileUpload darkMode={darkMode} type='file' />}
          />
        </FileUploadScrollContainer>
      </FileUploadContainer>
      <DynamicFormInputs inputs={BASIC_JOB_APP_INPUTS} form={form} setForm={setForm} />
      <Grid container justifyContent='flex-end'>
        <Grid item xs={12} md={6} lg={4}>
          <Link to='/app-complete'>
            <SubmitButton color='info' fullWidth size='large' variant='contained'>
              Submit Application
            </SubmitButton>
          </Link>
        </Grid>
      </Grid>
    </JobAppContainer>
  );
};

const JobAppContainer = styled(Grid)(({ darkMode }) => ({
  backgroundColor: darkMode ? '#303030' : '#F5F5F5',
  border: darkMode ? 'none' : '1px solid lightgrey',
  borderRadius: 10,
  boxShadow: darkMode ? '5px 3px 3px lightgrey' : '5px 3px 3px grey',
  marginLeft: '9%',
  marginTop: 25,
  padding: 60,
  textAlign: 'center',
  width: '80%',
}));

const JobAppTitleContainer = styled(Grid)({
  marginBottom: 30,
});

const JobAppTitleText = styled(Typography)(({ darkMode }) => ({
  color: darkMode ? 'beige' : 'inherit',
}));

const SubmitButton = styled(Button)({
  marginTop: 50,
});

const FileUploadContainer = styled(Grid)(({ darkMode }) => ({
  'h1,h2,h3,h4,h5,h6': {
    color: darkMode ? '#E8DED1' : 'inherit',
  },
  background: darkMode ? '#808080' : '#FAFAFA',
  border: darkMode ? 'none' : '1px dashed gainsboro',
  borderRadius: 6,
  boxShadow: darkMode ? '3px 2px 5px lightgrey' : '1px 1px 2px grey',
  marginBottom: 30,
  padding: 20,
}));

const FileUploadScrollContainer = styled(Grid)(({ darkMode }) => ({
  backgroundColor: darkMode ? '#8a8a8a' : '#f7f7f7',
  borderRadius: 5,
  marginTop: 20,
  maxHeight: 250,
  overflowY: 'auto',
  padding: 10,
}));

const FileUpload = styled(TextField)(({ darkMode }) => ({
  '.MuiInputBase-root': {
    backgroundColor: darkMode ? '#c7c7c7' : 'beige',
  },
}));
