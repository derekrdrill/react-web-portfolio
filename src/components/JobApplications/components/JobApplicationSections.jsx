import React, { useContext, useState } from 'react';
import styled from 'styled-components';
import { Grid, Typography } from '@mui/material';

import { DynamicFormInputs } from '../../DynamicFormInputs/DynamicFormInputs';
import { DynamicList } from '../../DynamicList/DynamicList';

import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { formInputsGenerator } from '../../../utils/formInputsGenerator';

export const JobApplicationSection = ({ section }) => {
  const { darkMode } = useContext(DarkLightModeContext);

  const [form, setForm] = useState(formInputsGenerator(section.inputs));

  const inputs = <DynamicFormInputs inputs={section.inputs} form={form} setForm={setForm} />;

  return (
    <JobAppSectionContainer container>
      <Grid item xs={12}>
        <JobAppSectionHeader darkMode={darkMode} variant='h4'>
          {section.title}
        </JobAppSectionHeader>
      </Grid>
      <Grid item xs={12}>
        <JobAppSectionHeader darkMode={darkMode} variant='subtitle1' component='span'>
          {section.instructions}
        </JobAppSectionHeader>
      </Grid>
      {section.dynamicList ? <DynamicList children={inputs} /> : inputs}
    </JobAppSectionContainer>
  );
};

const JobAppSectionContainer = styled(Grid)({
  padding: 20,
});

const JobAppSectionHeader = styled(Typography)(({ darkMode }) => ({
  color: darkMode ? '#fafafa' : 'inherit',
  paddingBottom: 10,
}));
