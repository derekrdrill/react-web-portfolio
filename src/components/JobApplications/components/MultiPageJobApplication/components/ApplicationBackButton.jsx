import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Button } from '@mui/material';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronLeft } from '@fortawesome/fontawesome-free-solid';

import { DarkLightModeContext } from '../../../../DarkLightMode/context/DarkLightModeContext';

export const getBackButtonTitle = (inputs, page) =>
  `\u00A0\u00A0${page > 1 ? inputs[page - 2].title : 'LANDING PAGE'}`;

export const ApplicationBackButton = ({ goBackward, inputs, page }) => {
  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <StyledButton color='info' darkMode={darkMode} onClick={goBackward}>
      <FontAwesomeIcon icon={faChevronLeft} />
      {getBackButtonTitle(inputs, page)}
    </StyledButton>
  );
};

ApplicationBackButton.propTypes = {
  goBackward: PropTypes.func,
  inputs: PropTypes.array,
  page: PropTypes.number,
};

export const StyledButton = styled(Button)(({ darkMode }) => ({
  ':hover': {
    color: darkMode && 'white',
  },
  marginTop: 15,
}));
