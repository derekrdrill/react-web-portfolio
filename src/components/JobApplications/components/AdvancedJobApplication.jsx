import React, { useContext } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { Grid, Typography, Button } from '@mui/material';

import { JobApplicationSection } from './JobApplicationSections';

import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { ADVANCED_JOB_APP_INPUTS } from '../constants/ADVANCED_JOB_APP_INPUTS';
import headerLogo from '../../../assets/header-logo.png';

export const AdvancedJobApplication = () => {
  const { darkMode } = useContext(DarkLightModeContext);

  return (
    <JobAppContainer item darkMode={darkMode}>
      <JobAppHeaderContainer container darkMode={darkMode}>
        <JobAppImageContainer item xs={12} md={3}>
          <Grid container justifyContent={{ xs: 'center', sm: 'flex-start' }}>
            <img src={headerLogo} />
          </Grid>
        </JobAppImageContainer>
        <JobAppTitleContainer item xs={12} md={9}>
          <Grid container justifyContent={{ xs: 'flex-start', md: 'flex-end' }}>
            <Typography textAlign={{ xs: 'center' }} variant='h3'>
              Application for Employment
            </Typography>
          </Grid>
        </JobAppTitleContainer>
        <JobAppInstructionsContainer container justifyContent='center'>
          <Typography component='span' textAlign={{ xs: 'center' }} variant='h6'>
            Please carefully read and answer all questions below. You will not be able to submit this application until
            all questions are answered.
          </Typography>
        </JobAppInstructionsContainer>
      </JobAppHeaderContainer>
      <div style={{ borderRight: '1px dashed black', borderLeft: '1px dashed black' }}>
        {ADVANCED_JOB_APP_INPUTS.map(section => (
          <JobApplicationSection key={section.id} section={section} />
        ))}
        <Grid container justifyContent='flex-end'>
          <JobAppSubmitButtonContainer item>
            <Link to='/app-complete'>
              <Button size='large' color='info' variant='contained'>
                Submit Application
              </Button>
            </Link>
          </JobAppSubmitButtonContainer>
        </Grid>
      </div>
    </JobAppContainer>
  );
};

const JobAppContainer = styled(Grid)(({ darkMode }) => ({
  borderRadius: 5,
  border: '1px solid lightgrey',
  backgroundColor: darkMode ? '#5c5c5c' : '#F5F5F5',
  boxShadow: '5px 3px 3px grey',
}));

const JobAppTitleContainer = styled(Grid)({
  padding: '30px 0',
});

const JobAppHeaderContainer = styled(Grid)(({ darkMode }) => ({
  backgroundColor: darkMode ? 'darkgrey' : 'gainsboro',
  borderRadius: '5px 5px 0 0',
  padding: 10,
}));

const JobAppSubmitButtonContainer = styled(Grid)({
  padding: 10,
});

const JobAppImageContainer = Grid;
const JobAppInstructionsContainer = Grid;
