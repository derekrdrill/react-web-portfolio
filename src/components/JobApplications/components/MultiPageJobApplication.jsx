import React, { useContext, useState } from 'react';
import { Link } from 'react-router-dom';
import styled, { createGlobalStyle } from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPenSquare, faLock, faChevronLeft, faChevronRight } from '@fortawesome/fontawesome-free-solid';
import { Button, IconButton, Grid, Typography, Toolbar } from '@mui/material';

import { JobApplicationSection } from './JobApplicationSections';
import { Breadcrumb as BreadcrumbComponent } from '../../Breadcrumb/Breadcrumb';
import { ProgressBar as ProgressBarComponent } from '../../ProgressBar/ProgressBar';

import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { ADVANCED_JOB_APP_INPUTS } from '../constants/ADVANCED_JOB_APP_INPUTS';

export const ProgressBar = ({ page, maxPage, progress }) =>
  page <= maxPage && <ProgressBarComponent progress={progress} page={page} />;

export const Breadcrumb = ({
  breadcrumbs,
  breadcrumbClick,
  darkMode,
  endCrumb,
  goBackward,
  goForward,
  maxPage,
  page,
  separator,
}) =>
  page >= 1 &&
  page <= maxPage && (
    <>
      <Grid container display={{ xs: 'none', sm: 'block' }}>
        <BreadcrumbComponent
          breadcrumbs={breadcrumbs}
          breadcrumbClick={breadcrumbClick}
          page={page}
          maxPage={maxPage}
          separator={separator}
          endCrumb={endCrumb}
        />
      </Grid>
      <Grid container>
        <Grid item xs={6}>
          <StyledButton darkMode={darkMode} onClick={goBackward}>
            <FontAwesomeIcon icon={faChevronLeft} />
            <BackToButtonTitle page={page} inputs={ADVANCED_JOB_APP_INPUTS} />
          </StyledButton>
        </Grid>
        <Grid item xs={6}>
          <Grid container justifyContent='flex-end'>
            <ForwardOrSubmitButton
              darkMode={darkMode}
              goForward={goForward}
              inputs={ADVANCED_JOB_APP_INPUTS}
              maxPage={maxPage}
              page={page}
            />
          </Grid>
        </Grid>
      </Grid>
    </>
  );

export const BackToButtonTitle = ({ page, inputs }) =>
  `\u00A0\u00A0BACK TO ${page > 1 ? inputs[page - 2].title : 'LANDING PAGE'}`;

export const ForwardButtonTitle = ({ page, maxPage, inputs }) =>
  `${page < maxPage - 1 ? `GO TO ${inputs[page].title}` : 'GO TO REVIEW'}\u00A0\u00A0`;

export const ForwardOrSubmitButton = ({ darkMode, page, maxPage, inputs, goForward }) =>
  page === maxPage ? (
    <StyledLink to='/app-complete'>
      <StyledButton darkMode={darkMode}>
        {`${'SUBMIT APPLICATION'}\u00A0\u00A0`}
        <FontAwesomeIcon icon={faChevronRight} />
      </StyledButton>
    </StyledLink>
  ) : (
    <StyledButton darkMode={darkMode} onClick={goForward}>
      <ForwardButtonTitle inputs={inputs} maxPage={maxPage} page={page} />
      <FontAwesomeIcon icon={faChevronRight} />
    </StyledButton>
  );

export const ApplicationContainer = ({ darkMode, inputs, page, maxPage, goForward, editMode, toggleEditMode }) => (
  <ApplicationContainerGrid container page={page} maxpage={maxPage}>
    {page < 1 ? (
      <Grid container className='landing-page'>
        <Grid item md={8}>
          <Typography variant='h4' component='h1'>
            Application for Employment
          </Typography>
          <Typography variant='subtitle1' component='h1'>
            Please carefully read and complete the following 8 section application
          </Typography>
          <Typography variant='subtitle1' component='h1'>
            <i>
              You can cancel at any time but will not be able to submit this application until all questions are
              answered
            </i>
          </Typography>
        </Grid>
        <Grid item xs={12}>
          <StartApplicationButton variant='contained' onClick={goForward}>
            Click here to begin the application
          </StartApplicationButton>
        </Grid>
      </Grid>
    ) : (
      <AdvancedAppInputsContainer
        className='app-inputs'
        darkMode={darkMode}
        item
        xs={12}
        editing={editMode}
        page={page}
        maxpage={maxPage}
      >
        <ReviewJobAppToolbar
          darkMode={darkMode}
          editMode={editMode}
          toggleEditMode={toggleEditMode}
          page={page}
          maxPage={maxPage}
        />
        {inputs.map(section => {
          return (
            <JobApplicationSectionContainer
              container
              key={section.id}
              page={page}
              sectionid={section.id}
              maxpage={maxPage}
            >
              <JobApplicationSection section={section} />
            </JobApplicationSectionContainer>
          );
        })}
      </AdvancedAppInputsContainer>
    )}
  </ApplicationContainerGrid>
);

export const ReviewJobAppToolbar = ({ darkMode, page, maxPage, editMode, toggleEditMode }) =>
  page === maxPage && (
    <StyledToolbar darkMode={darkMode}>
      <EditIcon onClick={toggleEditMode}>
        <ReviewJobAppToolbarIcon editMode={editMode} />
      </EditIcon>
    </StyledToolbar>
  );

export const ReviewJobAppToolbarIcon = ({ editMode }) => <FontAwesomeIcon icon={editMode ? faLock : faPenSquare} />;

export const MultiPageJobApplication = () => {
  const { darkMode } = useContext(DarkLightModeContext);

  const [editMode, setEditMode] = useState(null);
  const [progress, setProgress] = useState(0);
  const [page, setPage] = useState(0);
  const maxPage = 8;

  const goForward = () => {
    if (page < maxPage) {
      setPage(page + 1);
      setProgress(((page + 1) / maxPage) * 100);
    } else {
      setPage(maxPage + 1);
    }
  };

  const goBackward = () => {
    setPage(page - 1);
    setProgress(((page - 1) / maxPage) * 100);
  };

  const setPageByBreadcrumb = e => {
    let breadcrumbPage = Number(e.target.id);
    setPage(breadcrumbPage);
    setProgress((breadcrumbPage / maxPage) * 100);
  };

  const toggleEditMode = () => setEditMode(!editMode ? 'true' : null);

  return (
    <div>
      <PageBodyStyle darkMode={darkMode} page={page} maxpage={maxPage} />
      <ProgressBar page={page} maxPage={maxPage} progress={progress} />
      <Breadcrumb
        breadcrumbs={ADVANCED_JOB_APP_INPUTS}
        breadcrumbClick={setPageByBreadcrumb}
        endCrumb={'Review'}
        darkMode={darkMode}
        goBackward={goBackward}
        goForward={goForward}
        maxPage={maxPage}
        page={page}
        separator={'|'}
      />
      <ApplicationContainer
        darkMode={darkMode}
        inputs={ADVANCED_JOB_APP_INPUTS}
        page={page}
        maxPage={maxPage}
        goForward={goForward}
        editMode={editMode}
        toggleEditMode={toggleEditMode}
      />
    </div>
  );
};

export const PageBodyStyle = createGlobalStyle(({ darkMode, page }) => [
  page < 1 && {
    body: {
      backgroundImage: 'linear-gradient(to right, #851e38 , #e68fa4)',
    },
    'h1, h2, h3, h4, h5, h6': {
      color: 'beige',
    },
  },
  page >= 1 && {
    backgroundColor: darkMode ? 'black' : 'white',
  },
]);

export const ApplicationContainerGrid = styled(Grid)(({ page, maxpage }) => ({
  paddingTop: page < 1 ? 200 : page === maxpage ? 30 : 75,
  paddingLeft: page < 1 && 40,
}));

const StartApplicationButton = styled(Button)({
  backgroundColor: '#357960',
  ':hover': {
    backgroundColor: '#4dcb9d',
  },
});

export const AdvancedAppInputsContainer = styled(Grid)(({ darkMode, page, maxpage, editing }) => [
  page === maxpage && {
    borderColor: darkMode ? 'transparent' : 'lightgrey',
    borderRadius: 5,
    backgroundColor: darkMode ? '#2e2e2e' : '#F5F5F5',
    boxShadow: '5px 3px 3px grey',
    margin: '10px 30px 10px 30px',
    '.MuiButton-root': {
      pointerEvents: !editing && 'none',
      color: !editing && 'lightgrey',
    },
    '.MuiInputBase-input, .MuiInputBase-root': {
      backgroundColor: !editing && (darkMode ? 'darkgrey' : 'gainsboro'),
      pointerEvents: !editing && 'none',
    },
  },
]);

export const JobApplicationSectionContainer = styled(Grid)(({ page, sectionid, maxpage }) => [
  {
    display: 'none',
  },
  page === sectionid && {
    display: 'inline-block',
  },
  page === maxpage && {
    display: 'inline-block',
  },
]);

const StyledToolbar = styled(Toolbar)(({ darkMode }) => ({
  backgroundColor: darkMode ? 'grey' : 'gainsboro',
}));

const EditIcon = styled(IconButton)({
  path: {
    fill: 'darkslategrey',
    borderRadius: 5,
  },
});

const StyledButton = styled(Button)(({ darkMode }) => ({
  ':hover': {
    color: darkMode && 'white',
  },
  color: darkMode && '#cfcfcf',
  marginTop: 15,
}));

const StyledLink = styled(Link)({
  textDecoration: 'none',
});
