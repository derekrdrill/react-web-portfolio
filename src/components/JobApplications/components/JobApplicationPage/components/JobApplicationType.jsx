import React from 'react';
import PropTypes from 'prop-types';

import { BasicJobApplication } from '../../BasicJobApplication/BasicJobApplication';
import { AdvancedJobApplication } from '../../AdvancedJobApplication/AdvancedJobApplication';
import { MultiPageJobApplication } from '../../MultiPageJobApplication/MultiPageJobApplication';

export const JobApplicationType = ({ appType }) =>
  appType === 'basic' ? (
    <BasicJobApplication />
  ) : appType === 'advanced' ? (
    <AdvancedJobApplication />
  ) : (
    <MultiPageJobApplication />
  );

JobApplicationType.propTypes = {
  appType: PropTypes.string,
};
