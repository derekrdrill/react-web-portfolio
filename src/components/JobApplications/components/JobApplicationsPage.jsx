import React, { useState } from 'react';
import styled from 'styled-components';
import { Container, Row, Col } from 'react-bootstrap';
import { Select, InputLabel, MenuItem, FormControl, Typography } from '@mui/material';
import { BasicJobApplication } from './BasicJobApplication';
import { AdvancedJobApplication } from './AdvancedJobApplication';
import { MultiPageJobApplication } from './MultiPageJobApplication';
import { LoaderSpinner } from '../../LoaderSpinner/LoaderSpinner';

export const JobApplicationType = ({ appType }) =>
  appType === 'basic' ? (
    <BasicJobApplication />
  ) : appType === 'advanced' ? (
    <AdvancedJobApplication />
  ) : (
    <MultiPageJobApplication />
  );


export const JobApplicationsPage = () => {
  const [appType, setAppType] = useState('basic');
  const [loading, setLoading] = useState(null);

  const handleAppTypeChange = e => {
    setLoading(true);
    setTimeout(() => {
      setAppType(e.target.value);
      setLoading(null);
    }, 600);
  };

  return (
    <Container fluid>
      <JobApplicationsSelectRow>
        <Col xs={4} className='py-2'>
          <FormControl fullWidth>
            <InputLabel>
              <JobApplicationSelectLabelText variant='h6' component='label'>
                Select one
              </JobApplicationSelectLabelText>
            </InputLabel>
            <JobApplicationSelect value={appType} label='Select one' variant='standard' onChange={handleAppTypeChange}>
              <MenuItem value='basic'>Basic</MenuItem>
              <MenuItem value='advanced'>Advanced</MenuItem>
              <MenuItem value='multi-page'>Mulit-Page</MenuItem>
            </JobApplicationSelect>
          </FormControl>
        </Col>
      </JobApplicationsSelectRow>
      <Row>
        <JobApplicationType appType={appType} />
      </Row>
      <LoaderSpinner open={loading} />
    </Container>
  );
};

const JobApplicationsSelectRow = styled(Row)({
  backgroundColor: 'grey',
  position: '-webkit-sticky',
  position: 'sticky',
  top: 80,
  zIndex: 100,
});

const JobApplicationSelect = styled(Select)({
  backgroundColor: 'gainsboro',
  padding: 5,
  borderRadius: 5,
  marginLeft: 10,
});

const JobApplicationSelectLabelText = styled(Typography)({
  color: 'beige',
});
