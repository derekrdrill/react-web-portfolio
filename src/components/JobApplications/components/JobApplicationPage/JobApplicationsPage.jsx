import React, { useState } from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import { FormControl, Grid, InputLabel, MenuItem, Select, Typography } from '@mui/material';

import { JobApplicationType } from './components/JobApplicationType';
import { LoaderSpinner } from '../../../LoaderSpinner/LoaderSpinner';

import { DarkLightModeContext } from '../../../DarkLightMode/context/DarkLightModeContext';

export const JobApplicationsPage = () => {
  const { darkMode } = React.useContext(DarkLightModeContext);

  const [appType, setAppType] = useState('basic');
  const [loading, setLoading] = useState(null);

  const handleAppTypeChange = e => {
    setLoading(true);
    setTimeout(() => {
      setAppType(e.target.value);
      setLoading(null);
    }, 600);
  };

  return (
    <>
      <PageBodyStyle darkMode={darkMode} />
      <JobApplicationsSelectRow darkMode={darkMode} container>
        <Grid item xs={12} md={6} lg={4}>
          <FormControl fullWidth>
            <InputLabel>
              <JobApplicationSelectLabelText darkMode={darkMode} variant='h6' component='label'>
                Select one
              </JobApplicationSelectLabelText>
            </InputLabel>
            <JobApplicationSelect
              label='Select one'
              onChange={handleAppTypeChange}
              value={appType}
              variant='standard'
            >
              <MenuItem value='basic'>Basic</MenuItem>
              <MenuItem value='advanced'>Advanced</MenuItem>
              <MenuItem value='multi-page'>Mulit-Page</MenuItem>
            </JobApplicationSelect>
          </FormControl>
        </Grid>
      </JobApplicationsSelectRow>
      <Grid container>
        <Grid item xs={12}>
          <JobApplicationType appType={appType} />
        </Grid>
      </Grid>
      <LoaderSpinner open={loading} />
    </>
  );
};

export const PageBodyStyle = createGlobalStyle(({ darkMode }) => ({
  body: {
    backgroundColor: darkMode ? '#5c5c5c' : 'white',
  },
}));

export const JobApplicationsSelectRow = styled(Grid)(({ darkMode }) => ({
  backgroundColor: darkMode ? 'grey' : 'whitesmoke',
  borderTop: darkMode ? 'none' : '1px dashed #759CC9',
  padding: '20px 5px',
  position: 'sticky',
  top: 80,
  zIndex: 2,
}));

export const JobApplicationSelect = styled(Select)({
  backgroundColor: 'gainsboro',
  padding: 5,
  borderRadius: 5,
  marginLeft: 10,
});

export const JobApplicationSelectLabelText = styled(Typography)(({ darkMode }) => ({
  color: darkMode ? 'beige' : '#759CC9',
}));
