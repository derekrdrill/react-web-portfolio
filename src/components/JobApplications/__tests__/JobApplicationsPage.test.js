import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { JobApplicationsPage, JobApplicationType } from '../components/JobApplicationsPage';

const renderer = new ShallowRenderer();

describe('Job Applications Page tests', () => {
  it('renders correctly', () => {
    renderer.render(<JobApplicationsPage />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('renders type of job app type correctly', () => {
    const jobAppTypeTests = [
      { appType: 'basic', result: 'BasicJobApplication()' },
      { appType: 'advanced', result: 'AdvancedJobApplication()' },
      { appType: 'multi', result: 'MultiPageJobApplication()' },
    ];

    jobAppTypeTests.forEach(({ appType, result }) => {
      renderer.render(<JobApplicationType appType={appType} />);
      const render = renderer.getRenderOutput();
      const renderedAppType = render.type.toString();

      expect(renderedAppType).toContain(result);
    });
  });
});
