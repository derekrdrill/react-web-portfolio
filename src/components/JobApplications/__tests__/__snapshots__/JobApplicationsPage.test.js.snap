// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Applications Page tests renders correctly 1`] = `
<Container
  fluid={true}
>
  <Styled(Row)>
    <Col
      className="py-2"
      xs={4}
    >
      <ForwardRef(FormControl)
        fullWidth={true}
      >
        <ForwardRef(InputLabel)>
          <Styled(Component)
            component="label"
            variant="h6"
          >
            Select one
          </Styled(Component)>
        </ForwardRef(InputLabel)>
        <Styled(Component)
          label="Select one"
          onChange={[Function]}
          value="basic"
          variant="standard"
        >
          <ForwardRef(MenuItem)
            value="basic"
          >
            Basic
          </ForwardRef(MenuItem)>
          <ForwardRef(MenuItem)
            value="advanced"
          >
            Advanced
          </ForwardRef(MenuItem)>
          <ForwardRef(MenuItem)
            value="multi-page"
          >
            Mulit-Page
          </ForwardRef(MenuItem)>
        </Styled(Component)>
      </ForwardRef(FormControl)>
    </Col>
  </Styled(Row)>
  <Row>
    <JobApplicationType
      appType="basic"
    />
  </Row>
  <LoaderSpinner
    open={null}
  />
</Container>
`;
